### PR TITLE
Add a confirmation modal when deleting a year

### LINKED ISSUE #685: Confirmation Modal when deleting a year
When deleting a year, open a modal that makes the user confirm whether they want the year to be deleted.

### PR DESCRIPTION
**Pull request by Suraj-Ram**
# Description

When deleting a year, a modal opens for the user to confirm whether they want the year to be deleted.

https://github.com/sandboxnu/graduatenu/assets/30753067/645be1e1-c8f2-4230-8842-c86bdcfc3362

Closes #685 

## Type of change

Please tick the boxes that best match your changes.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] This requires a run of `yarn install`
- [ ] This has migration changes and requires a run of `yarn dev:migration:run`

# How Has This Been Tested?

Tested manually for account users and guests.

# Checklist:

- [x] I have run the production builds in docker for the frontend/backend and ensure things run fine. Check README of repo on how to run if not sure.
- [x] I have performed a self-review of my own code
- [ ] I have commented my code where needed
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I've run the end to end tests
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/frontend/hooks/useRedirectIfLoggedIn.ts [hunk 1]
```diff
@@ -15,22 +15,21 @@ export const useRedirectIfLoggedIn = () => {
...
-      router.push("/home");
-    } catch (err) {
-      const error = err as AxiosError;
-      logger.error(error);
-      setRenderSpinner(false);
-    }
-  };
-
   useEffect(() => {
+    const loginWithCookie = async () => {
```
Comment (Suraj-Ram) [inline packages/frontend/hooks/useRedirectIfLoggedIn.ts:19]: Moved function definition into `useEffect` to resolve linting warning

File: packages/frontend/components/Plan/DeleteYearModal.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  if (!student) {
+    return <></>;
+  }
+
+  const deleteYear = async () => {
+    try {
+      // refresh the cache, show success message, and close the modal
+      mutate(USE_STUDENT_WITH_PLANS_SWR_KEY);
+      //   setSelectedPlanId(null);
```
Comment (BrandonLim8890) [inline packages/frontend/components/Plan/DeleteYearModal.tsx:45]: Is there a reason why this is commented out?

File: packages/frontend/components/Plan/DeleteYearModal.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+  const { student } = useStudentWithPlans();
+
+  if (!student) {
+    return <></>;
+  }
+
+  const deleteYear = async () => {
+    try {
+      // refresh the cache, show success message, and close the modal
+      mutate(USE_STUDENT_WITH_PLANS_SWR_KEY);
```
Comment (BrandonLim8890) [inline packages/frontend/components/Plan/DeleteYearModal.tsx:44]: I would read up more on [SWR and Mutate](https://swr.vercel.app/docs/mutation#revalidation). Essentially the `mutate` should be called after the update of the plan, i.e after removing the year.

**Review by BrandonLim8890 — CHANGES_REQUESTED** [review]
Looks good so far!

File: packages/frontend/components/Plan/DeleteYearModal.tsx (hunk 3 above)
Comment (Suraj-Ram) [reply packages/frontend/components/Plan/DeleteYearModal.tsx:45]: Adressed in [7691ee0](https://github.com/sandboxnu/graduatenu/pull/689/commits/7691ee07923db58f72cb59a4d729f8c1cf2f4468).

File: packages/frontend/components/Plan/DeleteYearModal.tsx (hunk 2 above)
Comment (Suraj-Ram) [reply packages/frontend/components/Plan/DeleteYearModal.tsx]: Adressed in [7691ee0](https://github.com/sandboxnu/graduatenu/pull/689/commits/7691ee07923db58f72cb59a4d729f8c1cf2f4468).

**Review by KobeZ123 — APPROVED** [review]
👍

**Review by BrandonLim8890 — APPROVED** [review]

**Pull request merged**